internal/resources/metal/vlans/resource.go

```diff
@@ -0,0 +1,183 @@
+		s.Blocks = make(map[string]schema.Block)
+	}
+
+	resp.Schema = s
+}
+
+func (r *Resource) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
+	r.Meta.AddFwModuleToMetalUserAgent(ctx, request.ProviderMeta)
+	client := r.Meta.Metal
+
+	var data ResourceModel
+	response.Diagnostics.Append(request.Config.Get(ctx, &data)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	if data.Facility.IsNull() && data.Metro.IsNull() {
+		response.Diagnostics.AddError("Invalid input params",
+			equinix_errors.FriendlyError(errors.New("one of facility or metro must be configured")).Error())
+		return
+	}
+	if !data.Facility.IsNull() && !data.Vxlan.IsNull() {
+		response.Diagnostics.AddError("Invalid input params",
+			equinix_errors.FriendlyError(errors.New("you can set vxlan only for metro vlans")).Error())
+		return
+	}
+
+	createRequest := &packngo.VirtualNetworkCreateRequest{
+		ProjectID:   data.ProjectID.ValueString(),
+		Description: data.Description.ValueString(),
+	}
+	if !data.Metro.IsNull() {
+		createRequest.Metro = data.Metro.ValueString()
+		createRequest.VXLAN = int(data.Vxlan.ValueInt64())
+	}
+	if !data.Facility.IsNull() {
+		createRequest.Facility = data.Facility.ValueString()
+	}
+	vlan, _, err := client.ProjectVirtualNetworks.Create(createRequest)
+	if err != nil {
+		response.Diagnostics.AddError("Error creating Vlan", equinix_errors.FriendlyError(err).Error())
+		return
+	}
+
+	// Parse API response into the Terraform state
+	response.Diagnostics.Append(data.parse(vlan)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	// Set state to fully populated data
+	response.Diagnostics.Append(response.State.Set(ctx, &data)...)
+	return
+}
+
+func (r *Resource) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	r.Meta.AddFwModuleToMetalUserAgent(ctx, request.ProviderMeta)
+	client := r.Meta.Metal
+
+	var data ResourceModel
+	response.Diagnostics.Append(request.State.Get(ctx, &data)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	vlan, _, err := client.ProjectVirtualNetworks.Get(
+		data.ID.ValueString(),
+		&packngo.GetOptions{Includes: []string{"assigned_to"}},
+	)
+	if err != nil {
+		if equinix_errors.IsNotFound(err) {
+			response.Diagnostics.AddWarning(
+				"Equinix Metal Vlan not found during refresh",
+				fmt.Sprintf("[WARN] Vlan (%s) not found, removing from state", data.ID.ValueString()),
+			)
+			response.State.RemoveResource(ctx)
+			return
+		}
+		response.Diagnostics.AddError("Error fetching Vlan using vlanId",
+			equinix_errors.FriendlyError(err).Error())
+		return
+	}
+
+	response.Diagnostics.Append(data.parse(vlan)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	response.Diagnostics.Append(response.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Update(ctx context.Context, request resource.UpdateRequest, response *resource.UpdateResponse) {
+	//TODO implement me
+	panic("implement me")
```

Comment:
This should not panic.  See @displague's relevant comment on a different PR: https://github.com/equinix/terraform-provider-equinix/pull/588#discussion_r1502936323

Reply:
Yes removed panic